torch/nn/parallel/data_parallel.py

```diff
@@ -24,23 +31,27 @@ class DataParallel(Module):
         >>> output = net(input)
     """
 
-    def __init__(self, module, device_ids=None, output_device=None):
+    def __init__(self, module, device_ids=None, output_device=None, dim=0):
         super(DataParallel, self).__init__()
         if device_ids is None:
             device_ids = list(range(torch.cuda.device_count()))
         if output_device is None:
             output_device = device_ids[0]
+        self.dim = dim
         self.module = module
         self.device_ids = device_ids
         self.output_device = output_device
         if len(self.device_ids) == 1:
             self.module.cuda(device_ids[0])
 
     def forward(self, *inputs):
+        #leave non-Variable/non-tuples in place
         def _to_cuda(obj):
             if isinstance(obj, Variable):
                 return obj.cuda()
-            return tuple((map(_to_cuda, obj)))
+            if isinstance(obj, tuple):
+                return tuple((map(_to_cuda, obj)))
+            return obj
```

Comment:
I think a tensor might still need a call to `.cuda()` if `self.device_ids[0]` is different than its device (e.g. it's on CPU).